tests/unit/features/partitions/test_lifecycle_manager.py

```diff
@@ -0,0 +1,138 @@
+
+from string import ascii_lowercase
+from typing import Any, Dict
+
+import pytest
+from hypothesis import HealthCheck, given, settings, strategies
+
+from craft_parts import errors
+from craft_parts.lifecycle_manager import LifecycleManager
+
+
+class TestPartitionsSupport:
+    """Verify LifecycleManager supports partitions."""
+
+    @pytest.fixture
+    def parts_data(self) -> Dict[str, Any]:
+        return {"parts": {"foo": {"plugin": "nil"}}}
+
+    @pytest.mark.parametrize("partitions", [["default"], ["default", "kernel"]])
+    def test_project_info(self, new_dir, parts_data, partitions):
+        """Verify partitions are parsed and passed to ProjectInfo."""
+        lifecycle = LifecycleManager(
+            parts_data,
+            application_name="test_manager",
+            project_name="project",
+            cache_dir=new_dir,
+            work_dir="work_dir",
+            arch="aarch64",
+            parallel_build_count=16,
+            custom="foo",
+            partitions=partitions,
+        )
+        info = lifecycle.project_info
+
+        assert info.application_name == "test_manager"
+        assert info.project_name == "project"
+        assert info.target_arch == "arm64"
+        assert info.arch_triplet == "aarch64-linux-gnu"
+        assert info.parallel_build_count == 16
+        assert info.dirs.parts_dir == new_dir / "work_dir" / "parts"
+        assert info.dirs.stage_dir == new_dir / "work_dir" / "stage"
+        assert info.dirs.prime_dir == new_dir / "work_dir" / "prime"
+        assert info.custom_args == ["custom"]
+        assert info.custom == "foo"
+        assert info.partitions == partitions
+
+    @pytest.mark.parametrize("partitions", [None, []])
+    def test_no_partitions(self, new_dir, parts_data, partitions):
+        """Raise an error if the partitions feature is enabled but not defined."""
+        with pytest.raises(errors.FeatureError) as raised:
+            LifecycleManager(
+                parts_data,
+                application_name="test_manager",
+                cache_dir=new_dir,
+                partitions=partitions,
+            )
+
+        assert (
+            raised.value.message
+            == "Partition feature is enabled but no partitions are defined."
+        )
+
+    @pytest.mark.parametrize(
+        "partitions", [["defaulta"], ["kernel"], ["kernel", "default"]]
+    )
+    def test_default_not_first(self, new_dir, parts_data, partitions):
+        """Raise an error if the first partition is not 'default'."""
+        with pytest.raises(ValueError) as raised:
+            LifecycleManager(
+                parts_data,
+                application_name="test_manager",
+                cache_dir=new_dir,
+                partitions=partitions,
+            )
+
+        assert str(raised.value) == "First partition must be 'default'."
+
+    @settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
```

Comment:
Ahh yes, function-scoped fixtures. The bane of my hypothesis-usage...

These aren't a problem here for what you're doing here AFAICT, but strong caveat on disabling the health check since it won't detect additional fixtures added later, and then you'll get frustrated about the new fixture you add and why you're getting such weird results on this test, and you'll end up waking up at 2 AM and running to your laptop because you realise the problem.

Overly specific? Personal experience? Nah of course not!

Reply:
Ha!   I added a comment explaining why it is safe to suppress the health check here.

Reply:
Thanks! In general I do like to include human-readable explanations whenever I override something (I'm sure you've seen my additional comments when I #noqa something), so thank you for doing so!